ui/lib/pki/addon/components/pki-generate-root.js

```diff
@@ -0,0 +1,117 @@
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import errorMessage from 'vault/utils/error-message';
+
+/**
+ * @module PkiGenerateRoot
+ * PkiGenerateRoot shows only the fields valid for the generate root endpoint.
+ * This form handles the model save and rollback actions, and will call the passed
+ * onSave and onCancel args for transition (passed from parent).
+ * NOTE: this component is not TS because decorator-added parameters (eg validator and
+ * formFields) aren't recognized on the model.
+ *
+ * @example
+ * ```js
+ * <PkiGenerateRoot @model={{this.model}} @onCancel={{transition-to "vault.cluster"}} @onSave={{transition-to "vault.cluster.secrets"}} @adapterOptions={{hash actionType="import" useIssuer=false}} />
+ * ```
+ *
+ * @param {Object} model - pki/action model.
+ * @callback onCancel - Callback triggered when cancel button is clicked, after model is unloaded
+ * @callback onSave - Callback triggered after model save success.
+ * @param {Object} adapterOptions - object passed as adapterOptions on the model.save method
+ */
+export default class PkiGenerateRootComponent extends Component {
+  @service flashMessages;
+  @tracked showGroup = null;
+  @tracked modelValidations = null;
+  @tracked errorBanner = '';
+  @tracked invalidFormAlert = '';
+
+  @action
+  toggleGroup(group, isOpen) {
+    this.showGroup = isOpen ? group : null;
+  }
+
+  get defaultFields() {
```

Comment:
Having a component getter makes this so much clearer. Instead of passing a huge array of objects with arrays of attrs as `groupPropertyNames` to the model decorator (like we did in pki/roles)